packages/loader/driver-utils/src/summaryForCreateNew.ts

```diff
@@ -11,16 +12,56 @@ import {
 	IDocumentAttributes,
 } from "@fluidframework/protocol-definitions";
 
+/**
+ * Defines the current layout of an .app + .protocol summary tree
+ * this is used internally for create new, and single commit summary
+ * @internal
+ */
+export interface CombinedAppAndProtocolSummary extends ISummaryTree {
+	tree: {
+		[".app"]: ISummaryTree;
+		[".protocol"]: ISummaryTree;
+	};
+}
+
+/**
+ * Validates the current layout of an .app + .protocol summary tree
+ * this is used internally for create new, and single commit summary
+ * @internal
+ */
+export function isCombinedAppAndProtocolSummary(
+	summary: ISummaryTree | undefined,
+): summary is CombinedAppAndProtocolSummary {
+	if (
+		summary?.tree === undefined ||
+		summary.tree?.[".app"]?.type !== SummaryType.Tree ||
+		summary.tree?.[".protocol"]?.type !== SummaryType.Tree
+	) {
+		return false;
+	}
+	const treeKeys = Object.keys(summary.tree);
+	if (treeKeys.length !== 2) {
```

Comment:
Can't we have more than 2 and still have app and protocol summary? Where does the .logTail(I think there were more like this) lives? Are they within the app summary?

Reply:
this doesn't affect snapshots (download) only summaries(upload), and `.logTail` only exists in snapshost. everything managed by the runtime in under the `.app` tree. The summaries we validate here are always created by `combineAppAndProtocolSummary` which is just a few lines down in this file, and has a fixed format.